client/tx/tx.go

```diff
@@ -100,15 +101,20 @@ func BroadcastTx(clientCtx client.Context, txf Factory, msgs ...sdk.Msg) error {
 	}
 
 	if !clientCtx.SkipConfirm {
-		out, err := clientCtx.JSONMarshaler.MarshalJSON(tx)
+		msg, ok := tx.(proto.Message)
+		if !ok {
+			return fmt.Errorf("can't proto marshal %T", tx)
+		}
+
+		out, err := clientCtx.JSONMarshaler.MarshalJSON(msg)
```

Comment:
I think this should use `TxConfig.TxJSONEncoder`

Reply:
I think we can eliminate this check completely. We cannot use `TxConfig.TxJSONEncoder` on `msg` and we have `txBytes` error check for the `tx` here https://github.com/cosmos/cosmos-sdk/pull/7054/files#diff-df9cc9b2c6d08e86dbf0554ea3adb00aR130. @aaronc 



wdyt?

Reply:
this is handled in #7085

Reply:
fixed it anyway.